src/lighteval/tasks/templates/hellaswag.py

```diff
@@ -0,0 +1,152 @@
+
+from typing_extensions import NotRequired, TypedDict
+
+from lighteval.tasks.templates.continuation import get_continuation_prompt_function
+from lighteval.tasks.templates.multichoice import create_adapter_from_dict
+from lighteval.tasks.templates.utils.formatting_utils import (
+    capitalize,
+    fix_capitalization,
+    fix_ending_punct,
+    is_ended_sentence,
+)
+from lighteval.tasks.templates.utils.formulation import Formulation, MCFFormulation
+from lighteval.tasks.templates.utils.translation_literals import TRANSLATION_LITERALS
+from lighteval.utils.language import Language
+
+
+# NLI Cause/Effect (Copa)
+HELLASWAG_QUERY = "{activity_label}{ctx}"
+
+
+class HellaswagInput(TypedDict):
+    ctx_a: str
+    continuations: list[str]
+    gold_idx: int | list[int]
+    instruction: NotRequired[str]
+    activity_label: NotRequired[str]
+    ctx_b: NotRequired[str]
+
+
+class HellaswagAdapter(TypedDict):
+    ctx_a: str
+    continuations: str
+    gold_idx: str
+    instruction: NotRequired[str]
+    activity_label: NotRequired[str]
+    ctx_b: NotRequired[str]
+
+
+def get_hellaswag_prompt_function(
+    language: Language,
+    adapter: Callable[[dict], HellaswagInput] | HellaswagAdapter,
+    formulation: Formulation = MCFFormulation(),
+    dot_replacement: list[str] = [" [title]"],
+):
+    """
+    Create a templated prompt function for a Hellaswag task.
+
+    Format:
+    Context Premise thefore/cause | (Continuation 1, Continuation 2, Continuation 3)
+
+    Args:
+        language (Language): The language of the Hellaswag task.
+        adapter (Callable[[dict], HellaswagInput] | HellaswagAdapter): A function or dictionary to adapt the input data to the required HellaswagInput format.
+            Must map data from the dataset row to the HellaswagInput format.
+            Note: The gold_idx must be an index or list of indices in the continuations list, indicating the correct continuation(s).
+        formulation (Formulation, optional): The formulation to use for the task. Defaults to MCFFormulation().
+        dot_replacement (list[str], optional): A list of strings to replace with dot. We have to replace the the texts with dots because
+            of wikihow source.
+
+    Returns:
+        Callable: A function that generates COPA prompts based on the given parameters.
+    """
+
+    translation_literals = TRANSLATION_LITERALS[language]
+
+    def preprocess(text):
+        """Comes from AiHarness"""
+        # text = text.strip()
+        # NOTE: Brackets are artifacts of the WikiHow dataset portion of HellaSwag.
+        for dot_repl in dot_replacement:
+            text = text.replace(dot_repl, ". ")
+        text = re.sub("\\[.*?\\]", "", text)
+        text = text.replace("  ", " ")
+        text = text.replace(r"\.+", r"\.")
+        return text.strip()
+
+    def process_context(ctx):
+        if ctx == "":
+            return ""
+        return capitalize(fix_ending_punct(preprocess(ctx), translation_literals))
```

Comment:
Don't we already have those in lighteval?

Reply:
It would be better to have it once only if possible (especially since we already have some hellaswag definitions in the lib)